pkg/manifests/types.go

```diff
@@ -19,377 +19,398 @@ import (
+// `ThanosRulerConfig` defines configuration for the Thanos Ruler instance for user-defined projects.
 type ThanosRulerConfig struct {
-	// AlertmanagerConfigs holds configuration about how the Thanos Ruler component should communicate
-	// with aditional Alertmanager instances.
-	// default: nil
+	// Configures how the Thanos Ruler component communicates
+	// with additional Alertmanager instances.
+	// The default value is `nil`.
 	AlertmanagersConfigs []AdditionalAlertmanagerConfig `json:"additionalAlertmanagerConfigs,omitempty"`
-	// LogLevel defines the log level for Thanos Ruler.
-	// Possible values are: error, warn, info, debug.
-	// default: info
+	// Defines the log level setting for Thanos Ruler.
+	// The possible values are `error`, `warn`, `info`, and `debug`.
+	// The default value is `info`.
 	LogLevel string `json:"logLevel,omitempty"`
-	// NodeSelector defines which Nodes the Pods are scheduled on.
+	// Defines the nodes on which the Pods are scheduled.
 	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
-	// Resources define resources requests and limits for single Pods.
+	// Defines resource requests and limits for single pods.
 	Resources *v1.ResourceRequirements `json:"resources,omitempty"`
-	// Retention defines the time duration Thanos Ruler shall retain data for. Must match the regular expression
-	// [0-9]+(ms|s|m|h|d|w|y) (milliseconds seconds minutes hours days weeks years).
-	// default: 15d
+	// Defines the duration for which Prometheus retains data.
+	// This definition must be specified using the following regular
+	// expression pattern: `[0-9]+(ms|s|m|h|d|w|y)` (ms = milliseconds, 
+	// s= seconds,m = minutes, h = hours, d = days, w = weeks, y = years).
+	// The default value is `15d`.
 	Retention string `json:"retention,omitempty"`
-	// Tolerations defines the Pods tolerations.
+	// Defines tolerations for the pods.
 	Tolerations []v1.Toleration `json:"tolerations,omitempty"`
-	// TopologySpreadConstraints defines the pod's topology spread constraints.
+	// Defines topology spread constraints for the pods.
 	TopologySpreadConstraints []v1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
-	// VolumeClaimTemplate defines persistent storage for Thanos Ruler. It's possible to
-	// configure storageClass and size of volume.
+	// Defines persistent storage for Thanos Ruler. Use this setting to
+	// configure the storage class and size of a volume.
 	VolumeClaimTemplate *monv1.EmbeddedPersistentVolumeClaim `json:"volumeClaimTemplate,omitempty"`
 }
 
 // ----- Common Types -----
 
-// AdditionalAlertmanagerConfig defines configuration on how a component should communicate with
-// aditional Alertmanager instances.
+// `AdditionalAlertmanagerConfig` defines settings for how a component 
+// communicates with additional Alertmanager instances.
 type AdditionalAlertmanagerConfig struct {
-	// APIVersion defines the api version of Alertmanager.
+	// Defines the API version of Alertmanager.
```

Comment:
possible values are v1 or v2. Default is v2.

Reply:
fixed